spark-jobs/src/main/scala/filodb/downsampler/BatchDownsampler.scala

```diff
@@ -193,10 +194,13 @@ object BatchDownsampler extends StrictLogging with Instance {
         val bufferPool = offHeapMem.bufferPools(rawPartSchema.downsample.get.schemaHash)
         val downsamplers = chunkDownsamplersByRawSchemaId(rawSchemaId)
         val periodMarker = downsamplePeriodMarkersByRawSchemaId(rawSchemaId)
+
         val (_, partKeyPtr, _) = BinaryRegionLarge.allocateAndCopy(rawReadablePart.partKeyBase,
                                                    rawReadablePart.partKeyOffset,
                                                    offHeapMem.nativeMemoryManager)
 
+        RecordSchema.updateDSSchema(partKeyPtr, schemas)
```

Comment:
Call only if downsampleSchema != rawSchema ?

Reply:
Also add comment on why this is needed

Reply:
fixed